### PR TITLE
Adds versioning to the docs

### PR DESCRIPTION
**Pull request by sheikh-nv**
# Description

Adds versioning to the docs. Users can now choose a specific tag from version v1.0.0 onwards.

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -1,48 +1,55 @@
 name: Build & deploy docs
 
-on: [push]
+on:
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
on:
```

File: .github/workflows/docs.yaml [hunk 2]
```diff
@@ -1,48 +1,55 @@
...
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        echo exit code $?
+        git tag --list
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml:31]:
```suggestion
```

Comment (Dhoeller19) [reply .github/workflows/docs.yaml:31]: I guess these two are not necessary?

File: .github/workflows/docs.yaml [hunk 3]
```diff
@@ -1,48 +1,55 @@
...
+      working-directory: ./docs
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml:29]:
```suggestion
        git fetch --prune --unshallow origin main --tags
```

File: .github/workflows/docs.yaml [hunk 4]
```diff
@@ -1,48 +1,55 @@
...
+      working-directory: ./docs
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow origin main --tags
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        git fetch --prune --unshallow origin main --tags
        git branch -r
```

File: .github/workflows/docs.yaml [hunk 5]
```diff
@@ -1,48 +1,56 @@
...
+        git branch -r
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        git fetch --prune --unshallow origin main:refs/remotes/origin/main --tags
        git branch -r
```

File: .github/workflows/docs.yaml [hunk 6]
```diff
@@ -1,48 +1,56 @@
...
+      working-directory: ./docs
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow origin main:refs/remotes/origin/main --tags
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        git fetch --prune --unshallow origin main --tags
```

File: docs/conf.py [hunk 7]
```diff
@@ -240,7 +242,19 @@
     "icon_links_label": "Quick Links",
 }
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"^(main)$"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = r"^(main|origin/main)$"
```

File: .github/workflows/docs.yaml [hunk 8]
```diff
@@ -1,48 +1,56 @@
...
+      working-directory: ./docs
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow origin main --tags
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        git fetch --prune --unshallow --tags
```

File: docs/conf.py [hunk 9]
```diff
@@ -240,7 +242,19 @@
...
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"^(main|origin/main)$"
+
+# Whitelist pattern for tags (set to None to ignore all tags)
+smv_tag_whitelist = r"^v[1-9]\d*\.\d+\.\d+$"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_tag_whitelist = "tmp"  # r"^v[1-9]\d*\.\d+\.\d+$"
```

File: docs/conf.py [hunk 10]
```diff
@@ -240,7 +242,19 @@
     "icon_links_label": "Quick Links",
 }
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"^(main|origin/main)$"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = "pull/1247/merge"
```

File: docs/conf.py [hunk 11]
```diff
@@ -240,7 +242,19 @@
...
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = "pull/1247/merge"
+
+# Whitelist pattern for tags (set to None to ignore all tags)
+smv_tag_whitelist = r"^v[1-9]\d*\.\d+\.\d+$"
+
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
print(f"smv_branch_whitelist {smv_branch_whitelist}, smv_tag_whitelist {smv_tag_whitelist}")
```

File: docs/conf.py [hunk 12]
```diff
@@ -240,7 +242,19 @@
...
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = "pull/1247/merge"
+
+# Whitelist pattern for tags (set to None to ignore all tags)
+smv_tag_whitelist = r"^v[1-9]\d*\.\d+\.\d+$"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_tag_whitelist = "tmp"
```

File: docs/conf.py [hunk 13]
```diff
@@ -240,7 +242,19 @@
     "icon_links_label": "Quick Links",
 }
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = "pull/1247/merge"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = r"pull/1247/merge"
```

File: .github/workflows/docs.yaml [hunk 14]
```diff
@@ -1,48 +1,56 @@
...
+      working-directory: ./docs
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml:29]:
```suggestion
        git fetch --unshallow --tags
```

File: .github/workflows/docs.yaml [hunk 15]
```diff
@@ -1,48 +1,56 @@
...
+      working-directory: ./docs
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --unshallow --tags
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        git fetch --unshallow
```

File: .github/workflows/docs.yaml [hunk 16]
```diff
@@ -1,48 +1,56 @@
...
+      working-directory: ./docs
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --unshallow
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        git fetch --prune --unshallow --tags
```

File: docs/conf.py [hunk 17]
```diff
@@ -240,7 +242,19 @@
...
 }
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"pull/1247/merge"
+
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_remote_whitelist = r'^.*$'
```

File: docs/conf.py [hunk 18]
```diff
@@ -240,7 +242,19 @@
     "icon_links_label": "Quick Links",
 }
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"pull/1247/merge"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = r"^.*$"
```

File: docs/conf.py [hunk 19]
```diff
@@ -240,7 +242,19 @@
     "icon_links_label": "Quick Links",
 }
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"^.*$"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = r"^(main)$"
```

File: .github/workflows/docs.yaml [hunk 20]
```diff
@@ -1,48 +1,56 @@
...
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        git branch -r
+        echo exit code $?
+        git tag --list
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml:32]:
```suggestion
        git tag --list
        git branch --show-current
```

File: .github/workflows/docs.yaml [hunk 21]
```diff
@@ -1,48 +1,57 @@
...
+        git branch --show-current
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        echo "Current Branch: ${{ github.ref }}"
```

File: docs/conf.py (hunk 7 above)
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = r"^(refs/pull/1247/merge)$"
```

File: .github/workflows/docs.yaml [hunk 22]
```diff
@@ -1,48 +1,57 @@
...
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        git branch -r
+        echo exit code $?
+        git tag --list
+        echo "Current Branch: ${{ github.ref }}"
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml:33]:
```suggestion
        echo "Current branch: ${GITHUB_REF##*/}"
```

Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        echo "Current branch: $(git rev-parse --abbrev-ref HEAD)"
```

File: .github/workflows/docs.yaml [hunk 23]
```diff
@@ -1,48 +1,57 @@
...
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        git branch -r
+        echo exit code $?
+        git tag --list
+        echo "Current branch: $(git rev-parse --abbrev-ref HEAD)"
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}"
```

File: .github/workflows/docs.yaml [hunk 24]
```diff
@@ -1,48 +1,57 @@
...
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        git branch -r
+        echo exit code $?
+        git tag --list
+        echo "branch=${GITHUB_HEAD_REF:-${GITHUB_REF#refs/heads/}}"
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        echo "PR Source Branch: ${{ github.event.pull_request.head.ref }}"
```

File: docs/conf.py [hunk 25]
```diff
@@ -240,7 +242,19 @@
...
 }
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"^(refs/pull/1247/merge)$"
+smv_remote_whitelist = r'^.*$'
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = r"^(pull/1247/merge)$"
smv_remote_whitelist = r'^.*$'
```

File: docs/conf.py [hunk 26]
```diff
@@ -240,7 +242,19 @@
     "icon_links_label": "Quick Links",
 }
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"^(pull/1247/merge)$"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = r"^(refs/remotes/pull/1247/merge)$"
```

File: .github/workflows/docs.yaml [hunk 27]
```diff
@@ -1,48 +1,57 @@
...
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        git branch -r
+        echo exit code $?
+        git tag --list
+        echo "PR Source Branch: ${{ github.event.pull_request.head.ref }}"
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        echo "PR Source Branch: ${{ git branch 2> /dev/null | sed -e '/^[^*]/d' }}"
```

File: .github/workflows/docs.yaml [hunk 28]
```diff
@@ -1,48 +1,57 @@
...
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        git branch -r
+        echo exit code $?
+        git tag --list
+        echo "PR Source Branch: ${{ git branch 2> /dev/null | sed -e '/^[^*]/d' }}"
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
        current_branch=$(git branch 2> /dev/null | sed -e '/^[^*]/d' | awk '{print $2}')
        echo "Current branch: $current_branch"
```

File: docs/conf.py [hunk 29]
```diff
@@ -240,7 +242,19 @@
     "icon_links_label": "Quick Links",
 }
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"^(refs/remotes/pull/1247/merge)$"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = r"^(HEAD)$"
```

File: .github/workflows/docs.yaml [hunk 30]
```diff
@@ -1,48 +1,58 @@
...
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        git branch -r
+        echo exit code $?
+        git tag --list
+        current_branch=$(git branch 2> /dev/null | sed -e '/^[^*]/d' | awk '{print $2}')
+        echo "Current branch: $current_branch"
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
```

File: docs/conf.py [hunk 31]
```diff
@@ -240,7 +242,19 @@
...
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = r"^(HEAD)$"
+smv_remote_whitelist = r'^.*$'
+# Whitelist pattern for tags (set to None to ignore all tags)
+smv_tag_whitelist = "tmp"
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
# Whitelist pattern for remotes
smv_remote_whitelist = r"^.*$"
# Whitelist pattern for branches (set to None to ignore all branches)
smv_branch_whitelist = os.getenv("SMV_BRANCH_WHITELIST", r"^(main)$")
# Whitelist pattern for tags (set to None to ignore all tags)
smv_tag_whitelist = os.getenv("SMV_TAG_WHITELIST", r"^v[1-9]\d*\.\d+\.\d+$")
```

File: docs/conf.py [hunk 32]
```diff
@@ -240,7 +242,20 @@
...
+    "_templates",
+]
+
+# Whitelist pattern for remotes
+smv_remote_whitelist = r"^.*$"
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = os.getenv("SMV_BRANCH_WHITELIST", r"^(main)$")
+# Whitelist pattern for tags (set to None to ignore all tags)
+smv_tag_whitelist = os.getenv("SMV_TAG_WHITELIST", r"^v[1-9]\d*\.\d+\.\d+$")
+print(f"smv_branch_whitelist {smv_branch_whitelist}, smv_tag_whitelist {smv_tag_whitelist}")
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
```

File: .github/workflows/docs.yaml [hunk 33]
```diff
@@ -1,48 +1,56 @@
...
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        git branch -r
+        echo exit code $?
+        git tag --list
+        make build
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml:33]:
```suggestion
if [ -n "${secrets.SMV_BRANCH_WHITELIST}" ]; then
  make build SMV_BRANCH_WHITELIST=${secrets.SMV_BRANCH_WHITELIST}
else
  make build
fi
```

File: docs/conf.py [hunk 34]
```diff
@@ -240,7 +242,19 @@
...
 
-html_sidebars = {"**": ["navbar-logo.html", "icon-links.html", "search-field.html", "sbt-sidebar-nav.html"]}
+templates_path = [
+    "_templates",
+]
+
+# Whitelist pattern for remotes
+smv_remote_whitelist = r"^.*$"
+# Whitelist pattern for branches (set to None to ignore all branches)
+smv_branch_whitelist = os.getenv("SMV_BRANCH_WHITELIST", r"^(main)$")
```
Comment (Dhoeller19) [inline docs/conf.py]:
```suggestion
smv_branch_whitelist = os.getenv("SMV_BRANCH_WHITELIST", r"^(main|devel)$")
```

File: .github/workflows/docs.yaml [hunk 35]
```diff
@@ -1,48 +1,56 @@
...
-        working-directory: ./docs
-        run: pip install -r requirements.txt
+    - name: Install dev requirements
+      working-directory: ./docs
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Generate docs
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
    - name: Generate multi-version docs
```

File: docs/_redirect/index.html [hunk 36]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Redirecting to main branch</title>
+    <meta charset="utf-8">
+    <meta http-equiv="refresh" content="0; url=./main/index.html">
+    <link rel="canonical" href="https://isaac-sim.github.io/IsaacLab/main/index.html">
```
Comment (Dhoeller19) [inline docs/_redirect/index.html]:
```suggestion
```

File: docs/_redirect/index.html [hunk 37]
```diff
@@ -0,0 +1,8 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Redirecting to main branch</title>
```
Comment (Mayankm96) [inline docs/_redirect/index.html]:
```suggestion
    <title>Redirecting to the latest Isaac Lab documentation</title>
```

File: docs/conf.py [hunk 38]
```diff
@@ -213,7 +215,7 @@
     "show_toc_level": 1,
     "use_sidenotes": True,
     "logo": {
-        "text": "Isaac Lab documentation",
+        "text": "Isaac Lab Documentation",
```
Comment (Mayankm96) [inline docs/conf.py:218]:
```suggestion
        "text": f"Isaac Lab v{version}",
```

image

File: .github/workflows/docs.yaml [hunk 39]
```diff
@@ -1,48 +1,60 @@
...
-        with:
-          name: docs-html
-          path: ./docs/_build/html
+    - name: Generate multi-version docs
+      working-directory: ./docs
+      run: |
+        git fetch --prune --unshallow --tags
+        git branch -r
+        echo exit code $?
+        git tag --list
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
```

File: .github/workflows/docs.yaml [hunk 40]
```diff
@@ -1,48 +1,57 @@
...
+      working-directory: ./docs
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Check branch docs building
+      if: ${{ github.event_name == 'pull_request' }}
+      working-directory: ./docs
+      run: make single-docs
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
      run: |
        make single-docs
        rm -rf _build/latest
```

**Review by Dhoeller19 — APPROVED** [review]

File: .github/workflows/docs.yaml [hunk 41]
```diff
@@ -1,48 +1,58 @@
...
+      run: pip install -r requirements.txt
 
-      - name: Generate docs
-        working-directory: ./docs
-        run: make html
+    - name: Check branch docs building
+      if: ${{ github.event_name == 'pull_request' }}
+      working-directory: ./docs
+      run: |
+        make current-docs
```
Comment (Dhoeller19) [inline .github/workflows/docs.yaml]:
```suggestion
      run: make current-docs
```

**Pull request merged**